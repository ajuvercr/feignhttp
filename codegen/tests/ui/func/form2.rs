use feignhttp_codegen::post;

// error: two or more form parameters only supports scalar types, &str or String
//    |  pub async fn send_post(#[form] a: i32, #[form] b: &i32, #[form] f: F) -> Result<String> {}
//    |                                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

struct F;

#[post("http://xxx")]
pub async fn send_post(#[form] a: i32, #[form] b: &i32, #[form] f: F) -> Result<String> {}

fn main() {}
