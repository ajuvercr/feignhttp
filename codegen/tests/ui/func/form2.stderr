error: two or more form parameters only supports scalar types, &str or String
  --> tests/ui/func/form2.rs:10:32
   |
10 | pub async fn send_post(#[form] a: i32, #[form] b: &i32, #[form] f: F) -> Result<String> {}
   |                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
